### ISSUE TITLE
Not able to train network with Adagrad Solver.

### ISSUE THREAD

**Issue by rnsandeep** [BVLC/caffe#4068]
The parameters for the solver are:
net: "train_val.prototxt"             
test_iter: 5000                       
test_interval: 5000                   
test_initialization: false            
display: 1000                         
average_loss: 40                      
base_lr: 0.001                        
lr_policy: "fixed"                    
#power: 2.0                           
#gamma: 0.96                          
max_iter: 10000                       
weight_decay: 0.0002                  
snapshot: 2500                        
snapshot_prefix: "/disk1/classification/model/image_net_quick"
solver_mode: GPU                      
device_id: 2                          
type: "AdaGrad"

But the net is solving using sgd_solver.

The log file shows the following output.

I0428 02:33:17.797746 31652 solver.cpp:280] Learning Rate Policy: fixed
I0428 02:33:18.508793 31652 solver.cpp:228] Iteration 0, loss = 9.23844
I0428 02:33:18.508919 31652 solver.cpp:244]     Train net output #0: loss1/loss1 = 20.0148 (* 0.3 = 6.00443 loss)
I0428 02:33:18.508967 31652 solver.cpp:244]     Train net output #1: loss2/loss1 = 8.38504 (* 0.3 = 2.51551 loss)
I0428 02:33:18.508993 31652 solver.cpp:244]     Train net output #2: loss3/loss3_tune = 0.718495 (* 1 = 0.718495 loss)
I0428 02:33:18.509017 31652 **sgd_solver.cpp:106**] Iteration 0, lr = 0.001
I0428 02:44:05.604993 31652 solver.cpp:228] Iteration 1000, loss = 0.422412
I0428 02:44:05.605200 31652 solver.cpp:244]     Train net output #0: loss1/loss1 = 0.15632 (* 0.3 = 0.0468961 loss)
I0428 02:44:05.605213 31652 solver.cpp:244]     Train net output #1: loss2/loss1 = 0.194293 (* 0.3 = 0.0582878 loss)
I0428 02:44:05.605224 31652 solver.cpp:244]     Train net output #2: loss3/loss3_tune = 0.182052 (* 1 = 0.182052 loss)
I0428 02:44:05.605235 31652 **sgd_solver.cpp:106**] Iteration 1000, lr = 0.001

why the solver is not picking the adagrad but instead solving with sgd?

**Comment by seanbell** [comment]
That is the file name where the line is printed, not the solver name being used.  AdaGrad is a type of SGD solver.

e.g. you can see that the `AdaGrad` class derives from `SGDSolver`: https://github.com/BVLC/caffe/blob/master/include/caffe/sgd_solvers.hpp#L64

and line 106 is general: https://github.com/BVLC/caffe/blob/master/src/caffe/solvers/sgd_solver.cpp#L106

The only code that's unique to AdaGrad is `ComputeUpdateValue`: https://github.com/BVLC/caffe/blob/master/src/caffe/solvers/adagrad_solver.cpp#L14
